packages/back-end/src/util/migrations.ts

```diff
@@ -633,3 +639,53 @@ export function migrateSnapshot(
 
   return snapshot;
 }
+
+export type LegacySavedGroup = Omit<
+  SavedGroupInterface,
+  "source" | "condition"
+> & {
+  values?: string[];
+  source?: SavedGroupSource;
+  condition?: string;
+};
+function getGroupValues(values: string[], type?: string): string[] | number[] {
+  if (type === "number") {
+    return values.map((v) => parseFloat(v));
+  }
+  return values;
+}
+export function migrateSavedGroup(
+  doc: LegacySavedGroup,
+  attributes?: SDKAttributeSchema | undefined
+): SavedGroupInterface {
+  const attributeMap: AttributeMap = new Map();
+  attributes?.forEach((attribute) => {
+    attributeMap.set(attribute.property, attribute.datatype);
+  });
+
+  // Migrate missing fields - source and condition
+  const group: SavedGroupInterface = {
+    source: "inline",
+    condition: "",
+    ...doc,
+  };
+
+  // Migrate legacy values to new condition format
+  if (
+    group.source === "inline" &&
+    !group.condition &&
+    (group as LegacySavedGroup).values &&
+    group.attributeKey
+  ) {
+    group.condition = JSON.stringify({
+      [group.attributeKey]: {
+        $in: getGroupValues(
+          (group as LegacySavedGroup).values || [],
+          attributeMap.get(group.attributeKey)
```

Comment:
Is it possible that attributeMap no longer contains an attributeKey of an old saved group and if so does that affect a change in the code here?

Reply:
This PR actually fixes a bug here.  Previously, values were always stored as strings in Mongo and then swapped to numbers when generating the SDK payload depending on the attribute.  So if the attribute was deleted, the payload would switch to rendering strings (the default) instead of numbers.

Now, the correct data type is stored in Mongo, so even if the attribute is deleted, the payload will still render correctly.